pkg/api/dictionaryinfo.go

```diff
@@ -0,0 +1,338 @@
+	ItemValue string `json:"item_value"`
+	UpdatedAt string `json:"updated_at"`
+	CreatedAt string `json:"created_at"`
+}
+
+type DictionaryInfoCache struct {
+	client       *http.Client
+	token        string
+	logger       log.Logger
+	serviceCache *ServiceCache
+	enabled      bool
+
+	// Cached snapshot used by the collector to avoid network I/O on scrape.
+	snapshot []metricRow
+}
+
+type metricRow struct {
+	ServiceID      string
+	ServiceName    string
+	Version        int
+	DictionaryID   string
+	DictionaryName string
+	Digest         string
+	ItemCount      float64
+	LastUpdatedTS  float64
+}
+
+func NewDictionaryInfoCache(client *http.Client, token string, logger log.Logger, serviceCache *ServiceCache, enabled bool) *DictionaryInfoCache {
+	if logger == nil {
+		logger = log.NewNopLogger()
+	}
+	return &DictionaryInfoCache{
+		client:       client,
+		token:        token,
+		logger:       log.With(logger, "component", "dictionary-info"),
+		serviceCache: serviceCache,
+		enabled:      enabled,
+	}
+}
+
+func (c *DictionaryInfoCache) Enabled() bool { return c.enabled }
+
+// parse multiple time layouts commonly seen in Fastly APIs
+func parseFastlyTime(s string) (float64, bool) {
+	if s == "" {
+		return 0, false
+	}
+	layouts := []string{
+		time.RFC3339Nano,
+		time.RFC3339,
+		"2006-01-02 15:04:05",       // "2025-10-21 18:48:00"
+		"2006-01-02T15:04:05",       // without zone
+		"2006-01-02 15:04:05Z07:00", // space + zone
+	}
+	for _, layout := range layouts {
+		if t, err := time.Parse(layout, s); err == nil {
+			return float64(t.Unix()), true
+		}
+	}
+	return 0, false
+}
+
+// GET /service/{sid}/version/{v}/dictionary/{did}/item
+func (c *DictionaryInfoCache) listDictionaryItems(ctx context.Context, serviceID string, version int, dictID string) ([]dictItem, error) {
+	u := fmt.Sprintf("https://api.fastly.com/service/%s/version/%d/dictionary/%s/item", serviceID, version, dictID)
+	req, err := c.req(ctx, http.MethodGet, u)
+	if err != nil {
+		return nil, err
+	}
+	resp, err := c.client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+	if resp.StatusCode/100 != 2 {
+		return nil, fmt.Errorf("list dictionary items: %s", resp.Status)
+	}
+	var items []dictItem
+	return items, json.NewDecoder(resp.Body).Decode(&items)
+}
+
+// Refresh queries Fastly APIs and rebuilds the in-memory snapshot.
+func (c *DictionaryInfoCache) Refresh(ctx context.Context) error {
+	if !c.enabled {
+		return nil
+	}
+	var services []svc
+	if c.serviceCache != nil {
+		for _, s := range c.serviceCache.Services() {
+			active := s.Version
+			services = append(services, svc{
+				ID:   s.ID,
+				Name: s.Name,
+				Versions: []struct {
+					Number int  `json:"number"`
+					Active bool `json:"active"`
+				}{
+					{
+						Number: active,
+						Active: active > 0,
+					},
+				},
+			})
+		}
+	} else {
+		var err error
+		services, err = c.listServices(ctx)
+		if err != nil {
+			return err
+		}
+	}
```

Comment:
This else can be removed -- see above comment re: `c.serviceCache` which will always be not nil.